platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -93,3 +93,66 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 
 === Tags
 `#upgrades`
+
+[[favor_composite_builds]]
+== Favor Composite Builds over `buildSrc`
+:keywords: structuring-builds,composite-builds,buildSrc
+
+You should setup a <<composite_builds.adoc#defining_composite_builds,Composite Build>> (commonly called an "included build") to hold your _build logic_ (such as any plugins - including convention plugins - and other customizations that are specific to your build).
+
+=== Explanation
+
+The preferred location for build logic is an included build named `build-logic`, **not** in `buildSrc`.
+
+The automatically available `buildSrc` is great for rapid prototyping, but suffers from some subtle disadvantages:
+- There are classloader differences in how these 2 approaches behave that can be surprising; included builds are treated just like external dependencies, which is a simpler mental model.  Dependency resolution behaves subtly differently in buildSrc.
+- There can potentially be fewer task invalidations in a build when files in an included build are modified, leading to faster builds.
+- Included builds are complete Gradle builds and can be opened, worked on, and built independently as standalone projects.
+
+Ideally, there shouldn’t be any differences between these 2 approaches, and `buildSrc` should behave like an implicitly available included build if present.
+However, due to historical reasons, these differences currently exist.
+It is likely that in the future this recommendation will be weakened or removed entirely, as Gradle continues to work to converge the behavior of these two approaches.
+At present, however, the differences can be a potential stumbling block.
```

Comment:
Do we have any issues tracking the convergence? It might be nice to link.

Reply:
There is [this](https://github.com/gradle/gradle/issues/6045).  I'll link it in the references.